### PR TITLE
HPCC-29199 Remove SCMInterface only ECM files

### PR DESCRIPTION
**Pull request by kenrowland**
Removed SCMInterface only files and replaced with hpp files defining the interfaces currently in use

Signed-Off-By: Kenneth Rowland kenneth.rowland@lexisnexisrisk.com

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [x] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [x] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Review by afishbeck — COMMENTED** [review]
@kenrowland the changes look good, but I wonder if putting soapesp.hpp in esp/bindings would simplify things.  It's a header with interface definitions and covers general RPC even more than soap.  Then you could get rid of many of the CMakeLists.txt changes.  Long term I'd really like to stream line the entire directory structure.

**Comment by kenrowland** [comment]
@afishbeck I moved the file as discussed. The previously modified cmake files all had the necessary include directory for the new location. I know this commit shows a large number of modified files again. If you like I can rebase so that you only see the true modifications w/o all the noise.

File: dali/dfuplus/CMakeLists.txt [hunk 1]
```diff
@@ -69,5 +69,3 @@ target_link_libraries ( dfuplus
          xmllib
          esphttp
     )
-
```
Comment (afishbeck) [inline dali/dfuplus/CMakeLists.txt:72]: I would completely revert the changes and not include whitespace changes.  The whitespace cleanup is fine, but I wouldn't include it in this PR.

**Comment by afishbeck** [comment]
@kenrowland yes, go ahead and actually revert the files back to an untouched state and rebase.  I'm sure everything is fine but I'll review again after that.
